File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/inbound/web/credit/dto/UserCreditResponse.java
package com.study.webflux.rag.infrastructure.inbound.web.credit.dto;

import com.study.webflux.rag.domain.credit.model.UserCredit;

public record UserCreditResponse(
	String userId,
	long balance
) {
	public static UserCreditResponse from(UserCredit credit) {
		return new UserCreditResponse(credit.userId().value(), credit.balance());
	}
}

File: api/src/main/kotlin/com/miyou/app/api/credit/dto/UserCreditResponse.kt
package com.miyou.app.api.credit.dto

import com.miyou.app.domain.credit.model.UserCredit

data class UserCreditResponse(
    val userId: String,
    val balance: Long,
) {
    companion object {
        fun from(credit: UserCredit): UserCreditResponse = UserCreditResponse(credit.userId, credit.balance)
    }
}
